### ISSUE TITLE
Without-K doesn't prevent heterogeneous conflict between literals

### ISSUE THREAD

**Issue by jespercockx** [agda/agda#2721]
Agda accepts the following file:
```agda
{-# OPTIONS --without-K #-}

open import Agda.Primitive
open import Agda.Builtin.Nat
open import Agda.Builtin.Equality

data ⊥ : Set where

record Σ {a} {b} (A : Set a) (B : A → Set b) : Set (a ⊔ b) where
  constructor _,_
  field
    fst : A
    snd : B fst

no : _≡_ {A = Σ Set (λ X → X)} (Nat , 0) (Nat , 1) → ⊥
no ()
```
However, with univalence it's possible to construct a proof of `_≡_ {A = Σ Set (λ X → X)} (Nat , 0) (Nat , 1)`, so Agda shouldn't consider this type to be empty.

**Comment by jespercockx** [comment]
@asr can we still include this in the release? It's a straightforward fix and I don't expect it will break any code (though you never know).

**Comment by asr** [comment]
> @asr can we still include this in the release?

Yes. I'm waiting the fix of #2604 for "freezing" the stable branch.

Please add the issue to the list of fixed issues in the CHANGELOG file and close the issue after it.
